woke/setup.py

```diff
@@ -2,6 +2,7 @@
 
 install_requires = [
     "pydantic >= 1.9.0",
+    "typing_extensions >= 4.0, < 5",
```

Comment:
Should we put this in `extras_require['dev']` or `['tests']`?

Reply:
I think `['tests']` might be better?

Reply:
Should be in `install_requires`, I believe. Types `Literal` and `Annotated` are not part of Python 3.7 `typing` module so any import from `woke/woke/e_ast_parsing/b_solc/c_ast_nodes/__init__.py` causes an import error:
`ImportError: cannot import name 'Annotated' from 'typing' (/usr/lib/python3.7/typing.py)`

Reply:
I see. Correct me if I'm wrong: The behavior we want is:

- the program is executable on 3.7 (so we don't have to bump the Python version until it's really necessary)
- type checking also works on 3.7
- ideally we don't want type checking libs to be part of the executable

Does it accomplish these goals if we use conditionals? Something like

```python
from typing import TYPE_CHECKING
if TYPE_CHECKING:
  from typing_extensions import ...
```

Reply:
Is it safe to assume that 'type checking' only needs to be run in `test/dev` mode? I think so.. Interested to hear your opinion :-)

Reply:
> I see. Correct me if I'm wrong: The behavior we want is:
> 
> * the program is executable on 3.7 (so we don't have to bump the Python version until it's really necessary)

Yes

> * type checking also works on 3.7

Yes

> * ideally we don't want type checking libs to be part of the executable
> 
> Does it accomplish these goals if we use conditionals? Something like
> 
> ```python
> from typing import TYPE_CHECKING
> if TYPE_CHECKING:
>   from typing_extensions import ...
> ```

We have to distinguish between static type checking (MyPy and the other tools) and type checking at runtime (this is done by `pydantic`).
Static type checking should only be enforced in `test/dev`, sure.

But we also use `pydantic` to perform AST JSON validation at runtime. In my opinion, this also should be done in the executable (i.e. non-development environment). It is much better to tell the user "Oops we failed to parse this Solidity file" instead of silently ignoring the bug (= incompatible AST JSON).

We need types `Literal` and `Annotated` to perform `pydantic` validation.

Reply:
Ahhh ok